beacon-chain/core/state/interop_regression_test.go

```diff
@@ -55,7 +55,68 @@ func TestTopazRewardsRegression_StateRootValid(t *testing.T) {
 	// expect state transition to fail, as the block had an incorrect state root
 	postStateObj, err := ExecuteStateTransition(context.Background(), stateObj, blk)
 	if err == nil {
-		t.Error(err)
+		t.Error("State transition did not fail")
+	}
+	expectedRt, err := expectedStateObj.HashTreeRoot(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	rt, err := postStateObj.HashTreeRoot(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	if expectedRt != rt {
+		t.Errorf("Wanted a state root of %#x but got %#x", expectedRt, rt)
+	}
+	if !ssz.DeepEqual(expectedStateObj.InnerStateUnsafe(), postStateObj.InnerStateUnsafe()) {
+		diff, _ := messagediff.PrettyDiff(expectedStateObj.InnerStateUnsafe(), postStateObj.InnerStateUnsafe())
+		t.Errorf("Difference between states: %s", diff)
+	}
+}
+
+func TestSchlesiRewardsRegression_PyspecState(t *testing.T) {
+	regressionPath := "beacon-chain/core/state/regression_files"
+	blockFile := "schlesi_lighthouse_block.ssz"
```

Comment:
Why use light house block and pre state instead of the Prysm one?

Reply:
Prysm could never process this block since it was created by a lighthouse validator. So that is why its the lighthouse block. The pre state is assumed to be the same as prysm's for the purpose of the regression test. I dont have an easy way to get a ssz state from our debug endpoints right now and all prysm schlesi nodes are pretty much dead.

Reply:
Got it